kroxylicious-kms-provider-aws-kms-test-support/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsTestKekManager.java

```diff
@@ -0,0 +1,221 @@
+            checkForError(key, request.uri(), response.statusCode(), response);
+            return decodeJson(valueTypeRef, response.body());
+        }
+        catch (IOException e) {
+            if (e.getCause() instanceof KmsException ke) {
+                throw ke;
+            }
+            throw new UncheckedIOException("Request to %s failed".formatted(request), e);
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException("Interrupted during REST API call: %s".formatted(request.uri()), e);
+        }
+    }
+
+    private void checkForError(String key, URI uri, int statusCode, HttpResponse<byte[]> response) {
+        ErrorResponse error;
+        // AWS API states that only the 200 response is currently used.
+        // Our HTTP client is configured to follow redirects so 3xx responses are not expected here.
+        if (!isHttpSuccess(statusCode)) {
+            if (statusCode == 501) {
+                throw new AwsNotImplementException("AWS do not implement %s".formatted(uri));
+            }
+            try {
+                error = decodeJson(ERROR_RESPONSE_TYPE_REF, response.body());
+            }
+            catch (UncheckedIOException e) {
+                error = null;
+            }
+            if (error != null && error.isNotFound()) {
+                throw new UnknownAliasException(key);
+            }
+            throw new IllegalStateException("unexpected response %s (AWS error: %s) for request: %s".formatted(response.statusCode(), error, uri));
+        }
+    }
+
+    private void sendRequestExpectingNoResponse(HttpRequest request) {
+        try {
+            var response = client.send(request, HttpResponse.BodyHandlers.discarding());
+            if (!isHttpSuccess(response.statusCode())) {
+                if (response.statusCode() == 501) {
+                    throw new AwsNotImplementException("AWS do not implement %s".formatted(request.uri()));
+                }
+                throw new IllegalStateException("Unexpected response: %d to request %s".formatted(response.statusCode(), request.uri()));
+            }
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException("Request to %s failed".formatted(request), e);
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private boolean isHttpSuccess(int statusCode) {
+        return statusCode >= 200 && statusCode < 300;
+    }
+
+    private static <T> T decodeJson(TypeReference<T> valueTypeRef, byte[] bytes) {
+        try {
+            return OBJECT_MAPPER.readValue(bytes, valueTypeRef);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private String getBody(Object obj) {
```

Comment:
I'm aware this is probably just some shifting of old code, but I think a better name would be 
`renderBody` or `asJson` or `encodeJson` to give symmetry with `decodeJson`

Reply:
Yes, in fact I didn't touch any old code, just relocate the classes and remove the `exists` check for `deleteKek` and the other methods.

Agree to rename it.